### ISSUE TITLE
Trying to restore from backup, sidecar: index out of range error

### ISSUE THREAD

**Issue by george-angel** [coreos/etcd-operator#1613]
Error:

```
kubectl --context=dev -nbill-integration logs -f etcd-cluster-backup-sidecar-75c45f96fd-8lw7p
time="2017-11-01T17:38:14Z" level=info msg="listening on 0.0.0.0:19999"
time="2017-11-01T17:38:58Z" level=info msg="received a backup request"
time="2017-11-01T17:38:58Z" level=info msg="getMaxRev: member etcd-cluster-0028 revision (1)"
time="2017-11-01T17:38:58Z" level=info msg="skipped creating new backup: no change since last time"
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/coreos/etcd-operator/pkg/backup.(*Backup).Run(0xc4200dda40)
        /home/ubuntu/code/golang/src/github.com/coreos/etcd-operator/pkg/backup/backup.go:158 +0x30e
main.main()
        /home/ubuntu/code/golang/src/github.com/coreos/etcd-operator/cmd/backup/main.go:76 +0x32d
```

operator version `0.5.0`.

We couldn't restore from back, we deleted the cluster + sidecar + operator. The backup sidecar is in `CrashLoopBackOff` with this error.

**Comment by hongchaodeng** [comment]
This is a known issue that we fix after 0.5.0.
Please update etcd operator

**Comment by hongchaodeng** [comment]
Closing this as solved issue.

**Comment by george-angel** [comment]
Do you mind specifying the version which fixes this? We can't upgrade to `0.6.0` because of breaking contract changes.

**Comment by hongchaodeng** [comment]
Sorry. I mistake it as another similar issue.

I think we need to check the length in:
```
ack.status = bc.recentBackupsStatus[len(bc.recentBackupsStatus)-1]
```

We will get this fixed in next release 0.7.0 . We won't backport to 0.5.x. Sorry for that but we are still actively developing. If you can't upgrade please try to backport fix and build your custom image.

**Comment by george-angel** [comment]
No problem, we'll stay with `0.3.0` until we have the capacity to update our config to go to latest. Thanks.
